lib/jsonwp-proxy/proxy.js

```diff
@@ -135,8 +134,12 @@ class JWProxy {
       },
       resolveWithFullResponse: true,
       timeout: this.timeout,
-      forever: this.keepAlive,
     };
+    if (this.keepAlive) {
+      reqOpts.forever = true;
+    } else {
+      reqOpts.agent = false;
```

Comment:
Are these mutually exclusive? If in the initialization of the `reqOpts` object can it not be
```js
const reqOpts = {
  // ...
  forever: this.keepAlive,
  agent: !this.keepAlive,
}
```

Reply:
no it won't work this way, at least for `agent` key. This must be an instance of http.Agent class (`false` is only needed to disable it). Similar situation with `forever`: it implicitly sets the agent value if set to `true`, so we should not explicitly set its value on our own

Reply:
Ok! I think it might do well to be noted, otherwise at some point someone will do the 'optimization' I was thinking of.

Reply:
added comments